psycopg/psycopg/_pipeline.py

```diff
@@ -196,7 +197,7 @@ def __exit__(
         exc_tb: Optional[TracebackType],
     ) -> None:
         try:
-            self.sync()
+            self.sync(True)
```

Comment:
I would make this a kwarg-only argument, as `_sync_gen()` is,  and force user to use `fetch=True`.

Reply:
I'm not sure there is a use case for an explicit fetch so I'd rather rework the commit to not have the parameter in `sync()`. Do you see a use case on your side?

Reply:
No use case in mind. Just looking at that naked `True` in the `sync()`, which doesn't give any explanation, whereas the internal `_sync_gen()` is self explanatory with a kwargs-only argument :slightly_smiling_face: